common/debug/experiments.go

```diff
@@ -28,9 +28,30 @@ func IsIntermediateTrieCache() bool {
 	return intermediateTrieCache
 }
 
+// IsGetNodeData indicates whether the GetNodeData functionality should be enabled.
+// By default that's driven by the presence or absence of GET_NODE_DATA environment variable.
 func IsGetNodeData() bool {
-	gndEnv.Do(func() {
-		_, getNodeData = os.LookupEnv("GET_NODE_DATA")
-	})
-	return getNodeData
+	x := atomic.LoadUint32(&getNodeData)
+	if x >= 2 { // already initialized
+		return x%2 != 0
```

Comment:
Maybe we can use an enum instead of hardcoded values check. Like here it can be x&GetNodeDataFlag != 0 where const (
    GetNodeDataFlag = 1 << iota
    InitializedFlag
)